### PR TITLE
CISM support for NorESM diagnostics

### PR DESCRIPTION
**Pull request by hgoelzer**
This pull request adds basic CISM support to the NorESM diagnostics package. CISM diagnostic output can be selected as an additional component cism similar to the other components.

### CODE REVIEW

File: bin/diag_run [hunk 1]
```diff
@@ -384,6 +389,8 @@ else
             let RUN_MODELS[3]=${RUN_MODELS[3]}+1
         elif [ "$model" == "hamocc" ]; then
             let RUN_MODELS[4]=${RUN_MODELS[4]}+1
+        elif [ "$model" == "cism" ] || [ "$model" == "cism" ]; then
```
Comment (YanchunHe) [inline bin/diag_run]: change

```bash
elif [ "$model" == "cism" ] || [ "$model" == "cism" ]; then
```

to
```bash
elif [ "$model" == "cism" ]; then
```

??

File: bin/diag_run [hunk 2]
```diff
@@ -221,7 +223,7 @@ else
                     USER_MODEL=$1
                 else
                     echo "ERROR: no model specified (-m, --model)"
-                         echo "Valid options: cam, clm, cice, all"
+                         echo "Valid options: cam, clm, cice, cism, all"
```
Comment (YanchunHe) [inline bin/diag_run]: "Valid options: cam, blom/micom,clm, cice, cism, all"

File: packages/CAM_DIAG/amwg_template.csh [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/bin/csh
+#!/bin/csh -e
```
Comment (YanchunHe) [inline packages/CAM_DIAG/amwg_template.csh]: Usually, we want to keep the scrip running even if there are errors.
So I would suggest to remove '-e' option.

File: packages/CAM_DIAG/amwg_template.csh [hunk 4]
```diff
@@ -11,6 +11,7 @@ setenv DIAG_VERSION 140804  # version number YYMMDD
 if ( -d /opt/ncl65 && -d /opt/nco475 && -d /opt/cdo197 ) then
     setenv NCARG_ROOT /opt/ncl65
     setenv NCARG_COLORMAPS $NCARG_ROOT/lib/ncarg/colormaps
+#    setenv PATH /usr/bin:/usr/local/bin:/usr/bin:/opt/ncl65/bin/:/opt/nco475/bin/:/opt/cdo197/bin
```
Comment (YanchunHe) [inline packages/CAM_DIAG/amwg_template.csh]: I see this is for debug only, so the comment can be removed.

File: packages/CAM_DIAG/amwg_template.csh [hunk 5]
```diff
@@ -33,6 +34,10 @@ if ( $rcode != 0 ) then
     exit 1
 endif
 
+# hg workaround for cdo errors
+echo ncks: $ncksbin nco: $nco_dir cdo: $cdo_dir
+
```
Comment (YanchunHe) [inline packages/CAM_DIAG/amwg_template.csh]: clean up?

I don't there is problem with 'nco' path?

**Review by YanchunHe — CHANGES_REQUESTED** [review]
Some small revision of the code need to be done.

**Comment by hgoelzer** [comment]
Hi Yanchun,
I have reverted the changes in amwg_template.csh and corrected the issues you marked in diag_run. I hope the pull request can go forward now. Thanks
    Heiko

**Comment by YanchunHe** [comment]
Look good, I merge the changes. Thanks a lot, Heiko!

**Pull request merged**